src/Microsoft.AspNetCore.Blazor/UIEventArgs.cs

```diff
@@ -62,43 +193,330 @@ public class UIFocusEventArgs : UIEventArgs
+
+        /// <summary>
+        /// true if the meta key was down when the event was fired. false otherwise.
+        /// </summary>
+        public bool MetaKey { get; set; }
+    }
+
+    /// <summary>
+    /// Represents a single contact point on a touch-sensitive device.
+    /// The contact point is commonly a finger or stylus and the device may be a touchscreen or trackpad.
+    /// </summary>
+    public class UITouchPoint
+    {
+        /// <summary>
+        /// A unique identifier for this Touch object.
+        /// A given touch point (say, by a finger) will have the same identifier for the duration of its movement around the surface.
+        /// This lets you ensure that you're tracking the same touch all the time.
+        /// </summary>
+        public long Identifier { get; set; }
+
+        /// <summary>
+        /// The X coordinate of the touch point relative to the left edge of the screen.
+        /// </summary>
+        public long ScreenX { get; set; }
+
+        /// <summary>
+        /// The Y coordinate of the touch point relative to the top edge of the screen.
+        /// </summary>
+        public long ScreenY { get; set; }
+
+        /// <summary>
+        /// The X coordinate of the touch point relative to the left edge of the browser viewport, not including any scroll offset.
+        /// </summary>
+        public long ClientX { get; set; }
+
+        /// <summary>
+        /// The Y coordinate of the touch point relative to the top edge of the browser viewport, not including any scroll offset.
+        /// </summary>
+        public long ClientY { get; set; }
+
+        /// <summary>
+        /// The X coordinate of the touch point relative to the left edge of the document.
+        /// Unlike <see cref="ClientX"/>, this value includes the horizontal scroll offset, if any.
+        /// </summary>
+        public long PageX { get; set; }
+
+        /// <summary>
+        /// The Y coordinate of the touch point relative to the top of the document.
+        /// Unlike <see cref="ClientY"/>, this value includes the vertical scroll offset, if any.
+        /// </summary>
+        public long PageY { get; set; }
     }
 
     /// <summary>
     /// Supplies information about a mouse wheel event that is being raised.
     /// </summary>
-    public class UIWheelEventArgs : UIEventArgs
+    public class UIWheelEventArgs : UIMouseEventArgs
     {
+        /// <summary>
+        /// The horizontal scroll amount.
+        /// </summary>
+        public double DeltaX { get; set; }
+
+        /// <summary>
+        /// The vertical scroll amount.
+        /// </summary>
+        public double DeltaY { get; set; }
+
+        /// <summary>
+        /// The scroll amount for the z-axis.
+        /// </summary>
+        public double DeltaZ { get; set; }
+
+        /// <summary>
+        /// The unit of the delta values scroll amount. 
+        /// </summary>
+        public long DeltaMode { get; set; }
```

Comment:
This is really nice 👍

Reply:
Yeah... That data allow some _infinity scroll_ behavior to be implemented.